Rollbar.NetPlatformExtensions/RollbarLogger.cs

```diff
@@ -203,52 +202,44 @@ mslogging.EventId eventId
 
         private bool disposedValue = false; // To detect redundant calls
 
+
         /// <summary>
         /// Releases unmanaged and - optionally - managed resources.
         /// </summary>
         /// <param name="disposing">
         ///   <c>true</c> to release both managed and unmanaged resources; 
         ///   <c>false</c> to release only unmanaged resources.
         /// </param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Major Code Smell", "S1066:Collapsible \"if\" statements should be merged", Justification = "Cleaner code structure.")]
```

Comment:
Here, the S1066 error is being suppressed, but in Rollbar.OfflinePersistence/PayloadStoreRepository.cs the same code pattern is being used and doesn't require the error to be suppressed. Why is S1066 triggering here and not there? Or is this override unnecessary?

Reply:
looks like the warning is triggered only if there is more than one level of nested `if` conditions.

Reply:
The others do have two levels of nesting, but now I notice this one has three. It looks like that's the threshold, and I didn't notice the third level of nesting before.

Reply:
yes, that is easy to overlook and in general, I would take code analyzer suggestions with a grain of salt - while some suggestions definitely make sense and are absolutely valuable, others are just non-essential noise. One definitely shall use the suggestions as food for thought but not as a reason for unconditional action.